File: chia_doi.py
# def f(x):
#     # Định nghĩa hàm f(x) của phương trình đa thức bậc n
#     return 3*x**3 - x**2-1

# def chiadoi(a, b, epsilon):
#     # Sử dụng phương pháp chia đôi
#     while abs(b - a)/2 >= epsilon:       
#         c = (a + b) / 2
#         print(a,b,f(c))
#         if f(c) == 0:
#             return c
#         elif f(a) * f(c) < 0:
#             b = c
#         else:
#             a = c      
#     return c

# print("Nghiệm gần đúng của phương trình là:", chiadoi(0, 1, 0.001)) 


def f(x):
    # Định nghĩa hàm f(x) của phương trình đa thức bậc n
    return 3*x**3 - x**2-1

def chiadoi(a, b, esp):
    c = (a + b) / 2
    count=0
    while (abs(b - a)/2 > esp and f(c)!=0 and count <=5):       
        print(a,b,f(c))
        count+=count
        if f(a) * f(c) < 0:
            b = c
        else:
            a = c  
        c = (a + b) / 2
    return c

print("Nghiệm gần đúng của phương trình là:", chiadoi(0, 1, 0.001)) 

File: chiadoi.cpp


#include <iostream>
#include <cmath>
using namespace std;


//chia doi
float f(float x) {
    return 3*pow(x, 3) - 10*x +5;
}

float chiadoi(float a, float b) {
    float esp = 0.01;
    float c = (a + b) / 2;

    if (f(a) * f(b) < 0) {
        while (f(c) != 0 && fabs(a - b) > esp) {
            c = (a + b) / 2;           
            if (f(a) * f(c) < 0)
                b = c;
            else
                a = c;
        }
        return c;
    }
}

int main() {
    cout<<chiadoi(0, 1);   
}


//lap

//float f(float x) {
//    return pow(x+1, 1/3) ;
//}
//
//float lap(float a, float b) {
//    float esp = 0.01;
//    float x = a;
//    do {
//        float y = x;
//        x = f(x);
//    } while (abs(x - y), esp);
//    return x;
//}
//
//int main() {
//    cout<<lap(1, 2);
//}

//
//
//
//
//
//#define eps 0.01
//float f(float x) {
//    return pow(x + 1, 1.0 / 3.0);
//}
//
//int main() {
//    float x, y;
//    cout << "nhap x";
//    cin >> x;
//    do {
//        y = x;
//        x = f(x);
//        cout << x << y;
//    } while (fabs(x - y) > eps);
//        cout << x;
//}
